src/main/java/org/broadinstitute/dsm/route/PatchRoute.java

```diff
@@ -308,6 +314,52 @@ else if (Patch.DDP_PARTICIPANT_ID.equals(patch.getParent())) {
         }
     }
 
+    private void controlWorkflowByEmail(Patch patch, NameValue nameValue) {
+        try {
+            Map<String, String> pData = new Gson().fromJson(nameValue.getValue().toString(), Map.class);
+            DDPInstance ddpInstanceByGuid = Objects.requireNonNull(DDPInstance.getDDPInstanceByGuid(patch.getRealm()));
+            org.broadinstitute.dsm.model.participant.data.ParticipantData participantData =
+                    new org.broadinstitute.dsm.model.participant.data.ParticipantData(Integer.parseInt(patch.getId()),
+                            patch.getParentId(), Integer.parseInt(ddpInstanceByGuid.getDdpInstanceId()), patch.getFieldId(),
+                            pData);
+            if (participantData.hasFamilyMemberApplicantEmail()) {
+                int ddpInstanceIdByGuid = Integer.parseInt(ddpInstanceByGuid.getDdpInstanceId());
+                FieldSettings fieldSettings = new FieldSettings();
+                pData.forEach((columnName,columnValue) -> {
+                    if (!fieldSettings.isColumnExportable(ddpInstanceIdByGuid, columnName)) return;
+                    if (!patch.getFieldId().contains(org.broadinstitute.dsm.model.participant.data.ParticipantData.FIELD_TYPE)) return;
+                    ElasticSearchUtil.writeWorkflow(WorkflowForES.createInstanceWithStudySpecificData(ddpInstanceByGuid,
```

Comment:
Is there. a reason for not importing the `org.broadinstitute.dsm.model.participant.data.ParticipantData` once?

Reply:
Yes, there are two classes with the same name(ParticipantData) in different packages, PatchRoute uses both of them so to make differentiate which one is which one here is used full name of the class.